### ISSUE TITLE
Bela core does not compile

### ISSUE THREAD

**Issue by giuliomoro** [BelaPlatform/bela-image-builder#3]
<a href="https://github.com/LBDonovan"><img src="https://avatars1.githubusercontent.com/u/16209658?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [LBDonovan](https://github.com/LBDonovan)**
_Sunday Apr 16, 2017 at 17:54 GMT_
_Originally opened as https://github.com/LBDonovan/image-builder/issues/3_

----

clang fails on `math_runfast.c` with:

`clang: error: unknown argument: '-no-pie'`
`clang: error: unsupported argument '-mimplicit-it=arm' to option 'Wa,'`

gcc fails on Midi.cpp with:

`In file Midi.h: alsa/asoundlib.h: No such file or directory column: 28, line: 13`

**Comment by giuliomoro** [comment]
<a href="https://github.com/giuliomoro"><img src="https://avatars3.githubusercontent.com/u/4260604?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [giuliomoro](https://github.com/giuliomoro)**
_Monday Apr 24, 2017 at 13:02 GMT_

----

the `-no-pie`  is generated by `xeno-config`
you can manage to compile with
```
# remove dependencies from libasound:
# truncate files
echo > core/Midi.cpp
echo > core/Midi.h
# then remove `-lasound` from `Makefile`
```

At this point it compiles but it does not run:
it returns `-EINVAL` (invalid argument) when trying to do `rt_task_bind` or `rt_task_create`. Perhaps something went wrong in compiling libraries and tools ?

**Comment by giuliomoro** [comment]
<a href="https://github.com/giuliomoro"><img src="https://avatars3.githubusercontent.com/u/4260604?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [giuliomoro](https://github.com/giuliomoro)**
_Monday Apr 24, 2017 at 14:32 GMT_

----

`/usr/xenomai/bin/latency` does not depend on any of the xenomai lib (strange enough).
So I guess there may be something broken in the `libxenomai` build

**Comment by giuliomoro** [comment]
<a href="https://github.com/LBDonovan"><img src="https://avatars1.githubusercontent.com/u/16209658?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [LBDonovan](https://github.com/LBDonovan)**
_Monday Apr 24, 2017 at 14:38 GMT_

----

really? but /usr/xenomai/bin/xeno-test also runs fine...

**Comment by giuliomoro** [comment]
<a href="https://github.com/LBDonovan"><img src="https://avatars1.githubusercontent.com/u/16209658?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [LBDonovan](https://github.com/LBDonovan)**
_Monday Apr 24, 2017 at 14:39 GMT_

----

I did have to compile xenomai slightly oddly with '-f-no-pie' and '-no-pie' in CFLAGS and LDFLAGS, which did seem kinda wrong...

**Comment by giuliomoro** [comment]
<a href="https://github.com/giuliomoro"><img src="https://avatars3.githubusercontent.com/u/4260604?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [giuliomoro](https://github.com/giuliomoro)**
_Monday Apr 24, 2017 at 16:22 GMT_

----

actually none of `/usr/xenomai/bin/*` relies on `libxenomai`

**Comment by giuliomoro** [comment]
<a href="https://github.com/giuliomoro"><img src="https://avatars3.githubusercontent.com/u/4260604?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [giuliomoro](https://github.com/giuliomoro)**
_Tuesday Apr 25, 2017 at 01:44 GMT_

----

Got it to compile and run. I can even use the scope!

Recap:

* fix the xenomai lib (see #10)
* make sure the appropriate `.dtb` is loaded: add this line 
`dtb=am335x-bone-bela.dtb` 
somewhere in `/boot/uEnv.txt`
* compile with the `COMPILER=gcc` flag to `make`

**Comment by giuliomoro** [comment]
<a href="https://github.com/giuliomoro"><img src="https://avatars3.githubusercontent.com/u/4260604?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [giuliomoro](https://github.com/giuliomoro)**
_Wednesday Apr 26, 2017 at 00:09 GMT_

----

Boots and runs also on BBGW and BBBW.

Also hacked together a dtb for BBBW which makes the wifi module recognized. This is the diff between the stock BBBW and the BBBW that works for me (I basically diffed BBB vs BBG and tried to disable the same things (HDMI, some McASP) in the BBBW:
https://gist.github.com/giuliomoro/0b553fd15611a439c3eef6115b3e2122
